vignettes/Using_rerddap.Rmd

```diff
@@ -0,0 +1,671 @@
+buoyData$wspd <- as.numeric(buoyData$wspd)
+buoyData$time <- parse_date(buoyData$time)
+autoplot(xts(buoyData$wspd, order.by = buoyData$time)) + theme_bw() + ggtitle("Wind Speed in 2012 from buoy 46012")
+```
+
+
+###  IOOS Glider Data
+
+There is a repository for many of the gliders deployed in the oceans that is accessible through `rerddap` (http://data.ioos.us/gliders/erddap/).  Here we look at one glider deployed by Scripps.
+
+```{r glider, fig.width = 5, fig.height = 5, fig.align = 'center', warning = FALSE}
+  require("plot3D")
+require("rerddap")
+urlBase <- "https://data.ioos.us/gliders/erddap/"
+gliderInfo <- info("sp064-20161214T1913",  url = urlBase)
+glider <- tabledap(gliderInfo, fields = c("longitude", "latitude", "depth", "salinity"), 'time>=2016-12-14', 'time<=2016-12-23', url = urlBase)
+glider$longitude <- as.numeric(glider$longitude)
+glider$latitude <- as.numeric(glider$latitude)
+glider$depth <- as.numeric(glider$depth)
+scatter3D(x = glider$longitude , y = glider$latitude , z = -glider$depth, colvar = glider$salinity,              col = colors$salinity, phi = 40, theta = 25, bty = "g", type = "p", 
+           ticktype = "detailed", pch = 10, clim = c(33.2,34.31), clab = 'Salinity', 
+           xlab = "longitude", ylab = "latitude", zlab = "depth",
+           cex = c(0.5, 1, 1.5))
+```
+
+
+## Cacheing, "last", "now", idempotency, and a gotcha
+
+`rerddap` by default caches the requests you make, so that if you happen to make the same request again, the data is restored from the cache, rather than having to go out and retrieve it remotely.  For most applications, this a boon (such a when "knitting" and "reknitting" this document), as it can speed things up when doing a lot of request in a script, and works because in most cases an <span style="color:blue">ERDDAP</span> request is "idempotent".  This means that the the request will always return the same thing no matter what requests came before - it doesn't depend on state. However this is not true if the script uses either "last" in `griddap()` or "now" in `tabledap()` as these will return different values as time elapses and data are added to the datasets.  While it is desirable to have <span style="color:blue">ERDDAP</span> purely idempotent,  the "last" and "now" constructs are very helpful for people using <span style="color:blue">ERDDAP</span> in dashboards, webpages, regular input to models and the like, and the benefits far outweigh the problems.  However, if you are using either "last" or "now" in an `rerddap` based script, you want to be very careful to clear the `rerddap` cache, otherwise the request will be viewed as the same,  and the data from the last request, rather than the latest data, will be returned.  Note that several examples in this vignette use "last", and therefore the graphics may look different depending on when you "knitted" the vignette.
```

Comment:
We should probably add something to the documentation about this, will open an issue